src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -444,12 +432,74 @@ internal static string DecodeStream(Stream stream, Encoding encoding)
                     // Increment byteIndex to the next block of bytes in the input buffer, if any, to convert.
                     byteIndex += bytesUsed;
                 }
-            }
-            while (bytesRead != 0);
+            } while (bytesRead != 0);
 
             return result.ToString();
         }
 
+        internal static string DecodeStream(Stream stream, string characterSet, out Encoding encoding)
+        {
+            try
+            {
+                encoding = Encoding.GetEncoding(characterSet);
+            }
+            catch (ArgumentException)
+            {
+                encoding = null;
+            }
+            return DecodeStream(stream, ref encoding);
+        }
+
+        static bool TryGetEncoding(string characterSet, out Encoding encoding)
+        {
+            bool result = false;
+            try
+            {
+                encoding = Encoding.GetEncoding(characterSet);
+                result = true;
+            }
+            catch (ArgumentException)
+            {
+                encoding = null;
+            }
+            return result;
+        }
+
+        static readonly Regex s_metaexp = new Regex(@"<meta\s[.\n]*[^><]*charset\s*=\s*[""'\n]?(?<charset>[A-Za-z].[^\s""'\n<>]*)[\s""'\n>]");
+
+        internal static string DecodeStream(Stream stream, ref Encoding encoding)
+        {
+            bool isDefaultEncoding = false;
+            if (null == encoding)
+            {
+                // Use the default encoding if one wasn't provided
+                encoding = ContentHelper.GetDefaultEncoding();
+                isDefaultEncoding = true;
+            }
+
+            string content = StreamToString (stream, encoding);
+            if (isDefaultEncoding) do
+            {
+                // check for a charset attribute on the meta element to override the default.
+                Match match = s_metaexp.Match(content);
```

Comment:
@TravisEz13 - According to my reading of the page, I'm doing exactly what it says...

"To prevent recompilation, you should instantiate a single Regex object that is accessible to all code that requires it, as shown in the following rewritten example."

Since s_metaexp is static/readonly, there is only one compilation. That appears to be the point of the of the caching; avoiding recompilation.  Am I missing something?